### PR TITLE
cdc: add delay for recreating changefeed

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

Issue Number: ref #7657

### What is changed and how it works?

Delay create changefeed to workaround #7657.

#### Tests 

 - Unit test

##### Will it cause performance regression or break compatibility?

Yes, it might break compatibility in sense of recreate changefeed in a short time. 

##### Do you need to update user documentation, design documentation or monitoring documentation?

No.

### Release note 

```release-note
Add recreating changefeed peace period (30s) after a changefeed is removed.  
```

### CODE REVIEW

File: cdc/owner/owner.go [hunk 1]
```diff
@@ -285,6 +298,36 @@ func (o *ownerImpl) Query(query *Query, done chan<- error) {
...
+	t, ok := o.removedChangefeed[model.ChangeFeedID{ID: info.ID, Namespace: info.Namespace}]
+	if ok {
+		remain := recreateChangefeedDelayLimit - time.Since(t)
+		if remain >= 0 {
+			return cerror.ErrInternalServerError.GenWithStackByArgs(fmt.Sprintf(
+				"changefeed with same ID was just removed, please wait %s", remain))
+		}
+	}
+
+	sinkURI, err := url.Parse(info.SinkURI)
```
Comment (3AceShowHand) [inline cdc/owner/owner.go:314]: if `err` is not nil, also return nil ?

File: cdc/owner/owner.go [hunk 2]
```diff
@@ -50,6 +52,8 @@ const (
 // captures with versions different from that of the owner
 const versionInconsistentLogRate = 1
 
+var recreateChangefeedDelayLimit = 30 * time.Second
```
Comment (sdojjy) [inline cdc/owner/owner.go:55]: 30 seconds is too large and may impact the user experience, especially on the cloud, how about using a small delay, like 4 seconds?

**Review by asddongmen — APPROVED** [review]

File: cdc/owner/owner.go (hunk 2 above)
Comment (overvenus) [reply cdc/owner/owner.go:55]: It's a tradeoff between safety and UX, 4 seconds is too short, this issue is found where there are about 30s between delete and create again.

```
2022-11-19T01:40:30+08:00	{"level":"INFO","log":"[middleware.go:48] [/api/v2/changefeeds] [status=201] [method=POST] [path=/api/v2/changefeeds] [query=] [ip=10.233.86.4] [user-agent=Go-http-client/1.1] [client-version=v6.5.0-alpha] [] [duration=1.341204557s]","pod":"maincluster-ticdc-2","container":"ticdc","namespace":"stable-testbed-47l4r"}
2022-11-19T01:40:29+08:00	{"level":"INFO","log":"[middleware.go:48] [/api/v2/verify_table] [status=200] [method=POST] [path=/api/v2/verify_table] [query=] [ip=10.233.86.4] [user-agent=Go-http-client/1.1] [client-version=v6.5.0-alpha] [] [duration=819.420803ms]","pod":"maincluster-ticdc-2","container":"ticdc","namespace":"stable-testbed-47l4r"}
2022-11-19T01:40:28+08:00	{"level":"INFO","log":"[middleware.go:48] [/api/v2/tso] [status=200] [method=POST] [path=/api/v2/tso] [query=] [ip=10.233.86.4] [user-agent=Go-http-client/1.1] [client-version=v6.5.0-alpha] [] [duration=11.4233ms]","pod":"maincluster-ticdc-2","container":"ticdc","namespace":"stable-testbed-47l4r"}
2022-11-19T01:40:28+08:00	{"level":"INFO","log":"[middleware.go:48] [/api/v1/status] [status=200] [method=GET] [path=/api/v1/status] [query=] [ip=10.233.86.4] [user-agent=Go-http-client/1.1] [client-version=v6.5.0-alpha] [] [duration=40.921µs]","pod":"maincluster-ticdc-2","container":"ticdc","namespace":"stable-testbed-47l4r"}
2022-11-19T01:39:58+08:00	{"level":"INFO","log":"[middleware.go:48] [/api/v1/changefeeds/longrun-tidb-test-01] [status=400] [method=GET] [path=/api/v1/changefeeds/longrun-tidb-test-01] [query=] [ip=10.233.86.4] [user-agent=Go-http-client/1.1] [client-version=v6.5.0-alpha] [error=\"[CDC:ErrChangeFeedNotExists]changefeed not exists, {default longrun-tidb-test-01}\"] [errorVerbose=\"[CDC:ErrChangeFeedNotExists]changefeed not exists, {default longrun-tidb-test-01}\\ngithub.com/pingcap/errors.AddStack\\n\\tgithub.com/pingcap/errors@v0.11.5-0.20220729040631-518f63d66278/errors.go:174\\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\\n\\tgithub.com/pingcap/errors@v0.11.5-0.20220729040631-518f63d66278/normalize.go:164\\ngithub.com/pingcap/tiflow/cdc/owner.(*ownerStatusProvider).GetChangeFeedInfo\\n\\tgithub.com/pingcap/tiflow/cdc/owner/status_provider.go:126\\ngithub.com/pingcap/tiflow/cdc/api/v1.(*OpenAPI).GetChangefeed\\n\\tgithub.com/pingcap/tiflow/cdc/api/v1/api.go:209\\ngithub.com/gin-gonic/gin.(*Context).Next\\n\\tgithub.com/gin-gonic/gin@v1.7.4/context.go:165\\ngithub.com/pingcap/tiflow/cdc/api/middleware.ForwardToOwnerMiddleware.func1\\n\\tgithub.com/pingcap/tiflow/cdc/api/middleware/middleware.go:95\\ngithub.com/gin-gonic/gin.(*Context).Next\\n\\tgithub.com/gin-gonic/gin@v1.7.4/context.go:165\\ngithub.com/pingcap/tiflow/cdc/api/middleware.ErrorHandleMiddleware.func1\\n\\tgithub.com/pingcap/tiflow/cdc/api/middleware/middleware.go:64\\ngithub.com/gin-gonic/gin.(*Context).Next\\n\\tgithub.com/gin-gonic/gin@v1.7.4/context.go:165\\ngithub.com/pingcap/tiflow/cdc/api/middleware.LogMiddleware.func1\\n\\tgithub.com/pingcap/tiflow/cdc/api/middleware/middleware.go:38\\ngithub.com/gin-gonic/gin.(*Context).Next\\n\\tgithub.com/gin-gonic/gin@v1.7.4/context.go:165\\ngithub.com/pingcap/tiflow/cdc/api/middleware.CheckServerReadyMiddleware.func1\\n\\tgithub.com/pingcap/tiflow/cdc/api/middleware/middleware.go:103\\ngithub.com/gin-gonic/gin.(*Context).Next\\n\\tgithub.com/gin-gonic/gin@v1.7.4/context.go:165\\ngithub.com/gin-gonic/gin.CustomRecoveryWithWriter.func1\\n\\tgithub.com/gin-gonic/gin@v1.7.4/recovery.go:99\\ngithub.com/gin-gonic/gin.(*Context).Next\\n\\tgithub.com/gin-gonic/gin@v1.7.4/context.go:165\\ngithub.com/gin-gonic/gin.(*Engine).handleHTTPRequest\\n\\tgithub.com/gin-gonic/gin@v1.7.4/gin.go:489\\ngithub.com/gin-gonic/gin.(*Engine).ServeHTTP\\n\\tgithub.com/gin-gonic/gin@v1.7.4/gin.go:445\\nnet/http.serverHandler.ServeHTTP\\n\\tnet/http/server.go:2947\\nnet/http.(*conn).serve\\n\\tnet/http/server.go:1991\\nruntime.goexit\\n\\truntime/asm_amd64.s:1594\"] [duration=7.371972ms]","pod":"maincluster-ticdc-2","container":"ticdc","namespace":"stable-testbed-47l4r"}
2022-11-19T01:39:58+08:00	{"level":"INFO","log":"[middleware.go:48] [/api/v1/changefeeds/longrun-tidb-test-01] [status=202] [method=DELETE] [path=/api/v1/changefeeds/longrun-tidb-test-01] [query=] [ip=10.233.86.4] [user-agent=Go-http-client/1.1] [client-version=v6.5.0-alpha] [] [duration=296.241885ms]","pod":"maincluster-ticdc-2","container":"ticdc","namespace":"stable-testbed-47l4r"}
2022-11-19T01:39:58+08:00	{"level":"INFO","log":"[middleware.go:48] [/api/v1/changefeeds/longrun-tidb-test-01] [status=200] [method=GET] [path=/api/v1/changefeeds/longrun-tidb-test-01] [query=] [ip=10.233.86.4] [user-agent=Go-http-client/1.1] [client-version=v6.5.0-alpha] [] [duration=59.286755ms]","pod":"maincluster-ticdc-2","container":"ticdc","namespace":"stable-testbed-47l4r"}
2022-11-19T01:39:58+08:00	{"level":"INFO","log":"[middleware.go:48] [/api/v1/status] [status=200] [method=GET] [path=/api/v1/status] [query=] [ip=10.233.86.4] [user-agent=Go-http-client/1.1] [client-version=v6.5.0-alpha] [] [duration=41.001µs]","pod":"maincluster-ticdc-2","container":"ticdc","namespace":"stable-testbed-47l4r"}
2022-11-19T01:36:58+08:00	{"level":"INFO","log":"[middleware.go:48] [/api/v2/changefeeds] [status=201] [method=POST] [path=/api/v2/changefeeds] [query=] [ip=10.233.86.4] [user-agent=Go-http-client/1.1] [client-version=v6.5.0-alpha] [] [duration=675.878541ms]","pod":"maincluster-ticdc-2","container":"ticdc","namespace":"stable-testbed-47l4r"}
2022-11-19T01:36:58+08:00	{"level":"INFO","log":"[middleware.go:48] [/api/v2/verify_table] [status=200] [method=POST] [path=/api/v2/verify_table] [query=] [ip=10.233.86.4] [user-agent=Go-http-client/1.1] [client-version=v6.5.0-alpha] [] [duration=1.026702039s]","pod":"maincluster-ticdc-2","container":"ticdc","namespace":"stable-testbed-47l4r"}
2022-11-19T01:36:57+08:00	{"level":"INFO","log":"[middleware.go:48] [/api/v2/tso] [status=200] [method=POST] [path=/api/v2/tso] [query=] [ip=10.233.86.4] [user-agent=Go-http-client/1.1] [client-version=v6.5.0-alpha] [] [duration=200.084µs]","pod":"maincluster-ticdc-2","container":"ticdc","namespace":"stable-testbed-47l4r"}
2022-11-19T01:36:57+08:00	{"level":"INFO","log":"[middleware.go:48] [/api/v1/status] [status=200] [method=GET] [path=/api/v1/status] [query=] [ip=10.233.86.4] [user-agent=Go-http-client/1.1] [client-version=v6.5.0-alpha] [] [duration=40.472µs]","pod":"maincluster-ticdc-2","container":"ticdc","namespace":"stable-testbed-47l4r"}
```

**Review by sdojjy — APPROVED** [review]

**Comment by nongfushanquan** [comment]
/run-integration-test tidb=f5487e3f2596cdd63ae9840606c7fff3fb539e03
/run-integration-test

**Comment by nongfushanquan** [comment]
/run-integration-test tidb=f5487e3f2596cdd63ae9840606c7fff3fb539e03
/run-kafka-integration-test tidb=f5487e3f2596cdd63ae9840606c7fff3fb539e03

**Comment by overvenus** [comment]
/run-integration-test tidb=f5487e3f2596cdd63ae9840606c7fff3fb539e03
/run-kafka-integration-test tidb=f5487e3f2596cdd63ae9840606c7fff3fb539e03

**Comment by overvenus** [comment]
/run-kafka-integration-test tidb=f5487e3f2596cdd63ae9840606c7fff3fb539e03

**Pull request merged**